schema/pgsql/schema.sql

```diff
@@ -0,0 +1,1904 @@
+-- Icinga DB | (c) 2021 Icinga GmbH | GPLv2+
+
+CREATE DOMAIN bytea20 AS bytea CONSTRAINT exactly_20_bytes_long CHECK ( VALUE IS NULL OR octet_length(VALUE) = 20 );
+CREATE DOMAIN bytea16 AS bytea CONSTRAINT exactly_16_bytes_long CHECK ( VALUE IS NULL OR octet_length(VALUE) = 16 );
+CREATE DOMAIN bytea4 AS bytea CONSTRAINT exactly_4_bytes_long CHECK ( VALUE IS NULL OR octet_length(VALUE) = 4 );
+
+CREATE DOMAIN biguint AS bigint CONSTRAINT positive CHECK ( VALUE IS NULL OR 0 <= VALUE );
+CREATE DOMAIN uint AS bigint CONSTRAINT between_0_and_4294967295 CHECK ( VALUE IS NULL OR VALUE BETWEEN 0 AND 4294967295 );
+CREATE DOMAIN smalluint AS int CONSTRAINT between_0_and_65535 CHECK ( VALUE IS NULL OR VALUE BETWEEN 0 AND 65535 );
+CREATE DOMAIN tinyuint AS smallint CONSTRAINT between_0_and_255 CHECK ( VALUE IS NULL OR VALUE BETWEEN 0 AND 255 );
+
+CREATE TYPE boolenum AS ENUM ( 'n', 'y' );
+CREATE TYPE acked AS ENUM ( 'n', 'y', 'sticky' );
+CREATE TYPE state_type AS ENUM ( 'hard', 'soft' );
+CREATE TYPE checkable_type AS ENUM ( 'host', 'service' );
+CREATE TYPE comment_type AS ENUM ( 'comment', 'ack' );
+CREATE TYPE notification_type AS ENUM ( 'downtime_start', 'downtime_end', 'downtime_removed', 'custom', 'acknowledgement', 'problem', 'recovery', 'flapping_start', 'flapping_end' );
+CREATE TYPE history_type AS ENUM ( 'notification', 'state_change', 'downtime_start', 'downtime_end', 'comment_add', 'comment_remove', 'flapping_start', 'flapping_end', 'ack_set', 'ack_clear' );
+
+CREATE TABLE host (
+  id bytea20 NOT NULL,
+  environment_id bytea20 NOT NULL,
+  name_checksum bytea20 NOT NULL,
+  properties_checksum bytea20 NOT NULL,
+
+  name varchar(255) NOT NULL,
+  name_ci varchar(255) NOT NULL,
```

Comment:
That would require code in Web and Icinga DB to handle this column in a special way. I would not do that. PostgreSQL queries aren't that different.

Reply:
But wouldn't the current variant require the PostgreSQL SELECT queries do an extra `LOWER(name_ci)` anyways? Without having looked into this in too much detail, I think the `textci` type might give the behavior most similar to MySQL `COLLATE ..._ci`.

Reply:
Just wanted to say that I want to keep both columns. Sure, it would be perfect if we didn't need the `LOWER`. But if so, we can just add it if the column ends with `_ci`.

Reply:
> Sure, it would be perfect if we didn't need the `LOWER`. But if so, we can just add it if the column ends with `_ci`.

This would work for this column, but there are quite a few more that have no special naming but still are `COLLATE *_ci` in MySQL and have an index on `LOWER(col)` here.

Reply:
I doubt it would be much harder for Web to maintain a list of them. Or to even parse `SHOW CREATE TABLE`.